src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageProcessor.java

```diff
@@ -0,0 +1,203 @@
+                return new MessageProcessingResult(SUCCESS, ex);
+            }
+        } catch (Exception ex) {
+            logMessageProcessingError(message, payment, ex);
+            return new MessageProcessingResult(POTENTIALLY_RECOVERABLE_FAILURE);
+        }
+
+        return new MessageProcessingResult(POTENTIALLY_RECOVERABLE_FAILURE);
+    }
+
+    private void tryFinaliseProcessedMessage(IMessage message, MessageProcessingResult processingResult) {
+        try {
+            finaliseProcessedMessage(message, processingResult);
+        } catch (InterruptedException ex) {
+            logMessageFinaliseError(message, processingResult.resultType, ex);
+            Thread.currentThread().interrupt();
+        } catch (Exception ex) {
+            logMessageFinaliseError(message, processingResult.resultType, ex);
+        }
+    }
+
+    private void finaliseProcessedMessage(
+        IMessage message,
+        MessageProcessingResult processingResult
+    ) throws InterruptedException, ServiceBusException {
+
+        switch (processingResult.resultType) {
+            case SUCCESS:
+                messageReceiver.complete(message.getLockToken());
+                log.info("Message with ID {} has been completed", message.getMessageId());
+                break;
+            case UNRECOVERABLE_FAILURE:
+                deadLetterTheMessage(
+                    message,
+                    "Message processing error",
+                    processingResult.exception.getMessage()
+                );
+                break;
+            case POTENTIALLY_RECOVERABLE_FAILURE:
+                // starts from 0
+                int deliveryCount = (int) message.getDeliveryCount() + 1;
+
+                if (deliveryCount < maxDeliveryCount) {
+                    // do nothing - let the message lock expire
+                    log.info(
+                        "Allowing message with ID {} to return to queue (delivery attempt {})",
+                        message.getMessageId(),
+                        deliveryCount
+                    );
+                } else {
+                    deadLetterTheMessage(
+                        message,
+                        "Too many deliveries",
+                        "Reached limit of message delivery count of " + deliveryCount
+                    );
+                }
+
+                break;
+            default:
+                throw new MessageProcessingException(
+                    "Unknown message processing result type: " + processingResult.resultType
+                );
+        }
+    }
+
+    private void deadLetterTheMessage(
+        IMessage message,
+        String reason,
+        String description
+    ) throws InterruptedException, ServiceBusException {
+        messageReceiver.deadLetter(
+            message.getLockToken(),
+            reason,
+            description,
+            ImmutableMap.of("deadLetteredAt", Instant.now().toString())
```

Comment:
do we need this deadLetteredAt ? that is present in orchestrator for a specific reason as we use to delete the message based on this field.

Reply:
removed